### PR TITLE
iam_projects: drop projects column

### PR DESCRIPTION
**Pull request by srenatus**
### :nut_and_bolt: Description: What code changed, and why?

For some reason, we had wanted to assign a projects one or more projects. Doing that, we've introduced a `projects TEXT[]` database column, with no constraints whatsoever. Disliking that, we've discussed this again, and decided to drop the column. This is what this PR does.

### :chains: Related Resources

#832 

### :+1: Definition of Done

Our "projects" IAM entities no longer have "projects" themselves. There's no database column, and no API response, etc.

### :athletic_shoe: How to Build and Test the Change

- `start_all_services`, `rebuild components/authz-service`, `rebuild components/automate-gateway`
- check direct grpc calls:
```
[13][default:/src:0]# chef-automate dev grpcurl authz-service --  -d "$(jq -n '{id: "foo", name: "Foo" }')" chef.automate.domain.authz.v2.Projects.CreateProject
{
  "project": {
    "name": "Foo",
    "id": "foo",
    "type": "CUSTOM"
  }
}
[14][default:/src:0]# chef-automate dev grpcurl authz-service --  chef.automate.domain.authz.v2.Projects.ListProjects
{
  "projects": [
    {
      "name": "Foo",
      "id": "foo",
      "type": "CUSTOM"
    }
  ]
}
[15][default:/src:0]#
```
- check http requests:
```
[19][default:/src:0]# export TOKEN=$(chef-automate iam token create --admin iam-admin-token)
[20][default:/src:0]# curl -kH "api-token: $TOKEN" https://localhost/apis/iam/v2beta/projects/foo
{"project":{"name":"Foo","id":"foo","type":"CUSTOM"}}
[21][default:/src:0]# curl -kH "api-token: $TOKEN" https://localhost/apis/iam/v2beta/projects
{"projects":[{"name":"Foo","id":"foo","type":"CUSTOM"}]}
[22][default:/src:0]#
```

### :white_check_mark: Checklist

- [X] Tests added/updated?
- [ ] Docs added/updated?

### CODE REVIEW

File: components/authz-service/storage/v2/postgres/postgres.go [hunk 1]
```diff
@@ -1465,12 +1464,9 @@ func (p *pg) ListProjects(ctx context.Context) ([]*v2.Project, error) {
 }
 
 func (p *pg) insertProjectWithQuerier(ctx context.Context, project *v2.Project, q Querier) error {
-	_, err := q.ExecContext(ctx, `INSERT INTO iam_projects (id, name, type, projects)  VALUES ($1, $2, $3, $4);`,
-		project.ID, project.Name, project.Type.String(), pq.Array([]string{project.ID}))
-	if err != nil {
-		return err
-	}
-	return nil
+	_, err := q.ExecContext(ctx, `INSERT INTO iam_projects (id, name, type)  VALUES ($1, $2, $3);`,
```
Comment (stevendanna) [inline components/authz-service/storage/v2/postgres/postgres.go]: sorry:
```suggestion
	_, err := q.ExecContext(ctx, `INSERT INTO iam_projects (id, name, type) VALUES ($1, $2, $3);`,
```

Comment (srenatus) [reply components/authz-service/storage/v2/postgres/postgres.go]: Oh of course!

**Comment by srenatus** [comment]
> I assume we are OK with the backwards incompatible changes to the protobuf and sql files because this feature is in beta still?

Exactly! 😃

File: components/authz-service/storage/v2/postgres/postgres.go [hunk 2]
```diff
@@ -1374,8 +1374,8 @@ func (p *pg) UpdateProject(ctx context.Context, project *v2.Project) (*v2.Projec
 	// Update project if ID found AND intersection between projects and projectsFilter,
 	// unless the projectsFilter is empty (v2.0 case).
 	res, err := p.db.ExecContext(ctx,
-		`UPDATE iam_projects SET name =
-			$2 WHERE id = $1 AND (array_length($3::TEXT[], 1) IS NULL OR projects && $3);`,
+		`UPDATE iam_projects SET name=$2
+		WHERE id=$1 AND (array_length($3::TEXT[], 1) IS NULL OR id=ANY($3));`,
```
Comment (msorens) [inline components/authz-service/storage/v2/postgres/postgres.go]: Nice and tidy! ⭐️

File: inspec/a2-iam-v2p1-only-integration/controls/get_authorized_projects.rb [hunk 3]
```diff
@@ -89,6 +82,7 @@ def genInputProject(id, name)
               {
                 effect: "ALLOW",
                 # TODO this test will fail if there is a lone parameterized action (i.e. just iam:teams:get)
+                # ...what?! (@srenatus)
```
Comment (blakestier) [inline inspec/a2-iam-v2p1-only-integration/controls/get_authorized_projects.rb]: hmmmm . . . seems worth investigating

File: components/authz-service/storage/v2/postgres/postgres.go (hunk 2 above)
Comment (blakestier) [inline components/authz-service/storage/v2/postgres/postgres.go]: nvmd on my comment . . . found it here 🎉

File: inspec/a2-iam-v2p1-only-integration/controls/get_authorized_projects.rb (hunk 3 above)
Comment (srenatus) [reply inspec/a2-iam-v2p1-only-integration/controls/get_authorized_projects.rb]: Yup! Didn't get to it today.

Comment (srenatus) [reply inspec/a2-iam-v2p1-only-integration/controls/get_authorized_projects.rb]: https://github.com/chef/automate/pull/1109/commits/9b2105c124d702820b3674bbb7af447567ece5a1
not a problem. 🤷‍♂ I'll merge this.

**Pull request merged**